crates/rooch-framework/sources/auth_validator/ed25519_validator.move

```diff
@@ -79,10 +103,13 @@ module rooch_framework::ed25519_validator {
             auth_validator::error_invalid_account_auth_key()
         );
```

Comment:
Please help refactor the auth key checking mechanism. Right now it's only working for Ed25519 scheme since the address is generated from Ed25519. Need to consider other schemes that only have public and private keys.

Reply:
We have to update other auth key schemes `schnorr`, `ecdsa`, `ecdsa-recoverable` once to the account storage slots:

```move
let authentication_key = AuthenticationKey<ValidatorType> {
  authentication_key: new_auth_key,
};
account_storage::global_move_to(ctx, account, authentication_key);
```

Then we can check the auth key in the slot under the scenario like:

```move
let auth_key_option = account_authentication::get_authentication_key<Ed25519Validator>(ctx, addr);
if(option::is_some(&auth_key_option)){
   option::extract(&mut auth_key_option)
}
```